Doc/vnv/system test (#64)
dimitritsampiras/olympian: docs/VnVPlan/VnVPlan.tex

		
		Test Case Derivation: On a successful creation, the workout routine should be placed into the database.
		
		How test will be performed: This test can be preformed with a manual insertion of creating a workout routine under a test user. These results can be determined by viewing the database and scope of visibility of the created routine.
		
		Requirement(s): R1
		
		\item{\textbf{test-id1}}: A workout routine can be edited.
		
		Control: Manual
		
		Initial State: Application is running with an existing workout routine.
		
		Input: User edits a workout routine with new values.
		
		Output: A workout routine is updated with new values in the database and new values are visible to accessible users.
		
		Test Case Derivation: On a successful edit, the workout routine should be updated with new values database.
		
		How test will be performed: This test can be preformed with a manual edit of a workout routine under a test user.
		
		Requirement(s): R1
		
		\item{\textbf{test-id1}}: Adding an exercise to a workout routine.

		Control: Manual
		
		Initial State: Application is running and a workout routine is in the process of being created or edited.
		

> 		Input: a user created exercise with the parameters required for the exercise.

Reviewer: ```suggestion
		Input: A user created exercise with the parameters required for the exercise.
```

		
		Input: The user chooses to remove an exercise.
		
		Output: Some feedback indicating that an exercise has been removed. The workout routine no longer contains the removed exercise.
		
		Test Case Derivation: On removal of an exercise, there should be an indication towards the user notifying them of the removal. The exercise should also be deleted from the workout routine.
		
		How test will be performed: Given a workout routine, an exercise will be manually removed and checked to ensure that it no longer exists within the given workout routine.
		
		Requirement(s): R2
		
		\item{\textbf{test-id1}}: Limiting exercises to a workout routine.
		
		Control: Manual
		
		Initial State: Application is running and a workout routine is in the process of being created or edited.
		
		Input: The number of exercises required to reach the limit of exercises per workout routine. 
		
		Output: A notification to the user, before the limiting exercise notifying them of the limit, and preventing them from adding another exercise.
		
		Test Case Derivation: The user should be aware of the exercise limit per workout routine and should not be able to exceed it.
		

> 		How test will be performed: The test will be preformed by manually added exercises to a workout routine until the limit. Then upon adding the limiting input, a notification should be expected.

Reviewer: ```suggestion
		How test will be performed: The test will be preformed by manually added exercises to a workout routine until the limit. Then upon adding the limiting input, a notification is expected.
```

		Output: The exercise no longer holds a quantifier and is not displayed to the user.
		
		Test Case Derivation: On removing a quantifier to an exercise, it should no longer be visible to the user.
		
		How test will be performed: This test will be executed by removing a quantifier from an exercises to ensure it does not display any quantifying information.
		
		Requirement(s): R3
		
		\item{\textbf{test-id1}}: Editing quantifiers to an exercise.
		
		Control: Manual
		
		Initial State: The application is running and an exercise is in the process of being created or edited.
		
		Input: A number and unit of measurement describing an exercise.
		
		Output: The exercise now holds the updated quantifier.
		
		Test Case Derivation: On updating a quantifier to an exercise, there should be a new unit of measurement and/or value associated with it.
		
		How test will be performed: This test will be executed by editing a quantifier and ensuring the values are updated on completion of the edit.
		
		Requirement(s): R3
		
		\item{\textbf{test-id1}}: Publicizing a workout routine.
		
		Control: Manual
		
		Initial State: The application is running and a private workout routine is created.
		

> 		Input: An edit or addition to a workout routine to make is public.

Reviewer: ```suggestion
		Input: An edit or addition to a workout routine to make it public.
```

		
		Requirement(s): R5
		
		\item{\textbf{test-id1}}: Browsing workout routines.
		
		Control: Manual
		
		Initial State: The application is running and multiple workout routines are created.
		
		Input: navigation movements to view the public workout routines.
		
		Output: Multiple public workout routines should be displayed. 
		
		Test Case Derivation: On making a routine public, it should be available to all users and also the browsing page.
		
		How test will be performed: By first adding multiple public test routines, then checking the public workout routines by browsing. There should exist routines to access and view.
		
		Requirement(s): R6
		
		\item{\textbf{test-id1}}: Search workout routines.
		
		Control: Manual
		
		Initial State: The application is running and multiple workout routines are created.
		
		Input: Search string inputs to view the public workout routines.
		
		Output: Public workout routines that match the search criteria should be displayed. 
		

> 		Test Case Derivation: All workout routine that follow match a search, should be display.

Reviewer: ```suggestion
		Test Case Derivation: All workout routines that match a search, should be displayed.
```
